docs/general.md

```diff
@@ -0,0 +1,133 @@
+   [`@debugr/http-common`]
+ - `EntryType.GraphqlQuery` - used to denote a `GraphqlQueryLogEntry`, see
+   [`@debugr/graphql-common`]
+ - `EntryType.SqlQuery` - used to denote a `SqlQueryLogEntry`, see
+   [`@debugr/sql-common`]
+
+If you're writing a plugin which produces or consumes any of these entry types,
+you should add a dependency on the appropriate `@debugr/*-common` package
+and reuse the existing types if at all possible, rather than define your own types.
+
+## Plugin types
+
+Plugins in Debugr must conform to the `Plugin` interface exported from `@debugr/core`
+or (more probably) one of its descendants. The base `Plugin` interface defines only
+three things:
+
+ - `public readonly id: string`: An identifier of the plugin. This must be unique
+   to your plugin, otherwise it may lead to conflicts between plugin packages.
+ - `public readonly kind: PluginKind`: A discriminator property denoting the plugin type.
+   The `PluginKind` enum is a `const enum`, meaning you can define your own custom plugin
+   types similarly to how you can define custom log levels and entry types. Like with
+   entry types, a `PluginKind` value is associated with an interface derived from the
+   generic `Plugin` interface, by convention named `<Type>Plugin` (e.g. `CollectorPlugin`).
+   The built-in plugin types also provide type guards following the `is<Type>Plugin()` convention.
+   If you want to define your own plugin type, it is highly recommended you stick to this
+   convention.
+ - `public injectLogger(logger: Logger, pluginManager: PluginManager): void`: This method
+   will be called during Debugr initialisation. It allows plugins to get an instance of
+   the `Logger`, as well as check which other plugins are registered in the plugin manager
+   and optionally register more plugins they may require to work. More on this later.
+
+There are three predefined plugin types at this time:
+
+ - [**Collectors**]: Plugins which integrate Debugr with some kind of third party library
+   or framework. They collect any relevant log entries from the 3rd party code and if applicable
+   they may also start Debugr tasks for anything the library does which might be considered
+   a task from Debugr's point of view.
+ - [**Handlers**]: Plugins which consume log entries and persist them to some kind of storage.
+   They may optionally employ Formatter plugins to convert entries to the storage format.
+ - [**Formatters**]: These plugins convert log entries of a given type to a format expected
+   by a given Handler.
+
+## Data flow
+
+The following diagram demonstrates how data flows inside Debugr:
+
+```mermaid
+flowchart LR
+  c1(Collector 1) --> ll(Logger)
+  c2(Collector 2) --> ll
+  c3(Collector 2) --> ll
+  c4(User logs) --> ll
+  ll --> h1(Handler 1)
+  ll --> h2(Handler 2)
+  h1 --> s1[(Storage)]
+  h2 --> s2[(Storage)]
+  h2 <--> f2(Formatter)
+```
+
+
+[`@debugr/http-common`]: ../packages/http-common
```

Comment:
Tohle by se asi vyplatilo trosku popsat, nebo samostatny nadpis minimalne :)

Reply:
to je jen markdown syntax pro specifikovani URL odkazu, aby nemuselo bejt napsany primo v textu - ve vystupu to neni

Reply:
jojo, ale nejaky popisky nebo tak

Reply:
no ty jsou v textu právě

Reply:
tohle readme nikdo nebude konzumovat v raw md, všichni ho budou číst na GitHubu vyrenderovaný, a tam tyhle definice nebudou vidět - jen ty samotný odkazy v textu který je používaj